compute_sdk/globus_compute_sdk/sdk/executor.py

```diff
@@ -640,7 +685,12 @@ def _task_submitter_impl(self) -> None:
                     bs = max(1, self.batch_size)  # May have changed while waiting
                     while task is not None:
                         assert fut is not None  # Come on mypy; contextually clear!
-                        submit_group = task.task_group_uuid, task.endpoint_uuid
+                        submit_group = (
+                            task.task_group_uuid,
+                            task.endpoint_uuid,
+                            # dict type is unhashable
+                            json.dumps(task.user_endpoint_config),
+                        )
```

Comment:
Oof; this is serialized for every single task.  That's nominally "expensive," but I don't see a way around this in the current the current construction.

The more formal snapshotting idea we've recently discussed is looking to be more and more necessary, and perhaps sooner than expected.

Reply:
From a grouping standpoint, I expect that we're probably "fine," but we may want to consider sorting the keys.  That would provide a slightly stronger "guarantee" (?!?!) that the dictionary key would match more (all?) of the tasks in the logical submit grouping.

Reply:
Yeah, it's definitely not pretty, but I think we should implement (soon) the snapshot in a subsequent PR.